src/pretalx/schedule/models/slot.py

```diff
@@ -71,3 +74,22 @@ def copy_to_schedule(self, new_schedule, save=True):
         if save:
             new_slot.save()
         return new_slot
+
+    def build_ical(self, calendar, creation_time=None, netloc=None):
+        creation_time = creation_time or datetime.now(pytz.utc)
+        netloc = netloc or urlparse(settings.SITE_URL).netloc
+        tz = pytz.timezone(self.submission.event.timezone)
+
+        vevent = calendar.add('vevent')
+        vevent.add('summary').value = f'{self.submission.title} - {self.submission.display_speaker_names}'
+        vevent.add('dtstamp').value = creation_time
+        vevent.add('location').value = str(self.room.name)
+        vevent.add('uid').value = 'pretalx-{}-{}@{}'.format(
+            self.submission.event.slug, self.submission.code,
+            netloc
+        )
+
+        vevent.add('dtstart').value = self.start.astimezone(tz)
+        vevent.add('dtend').value = self.end.astimezone(tz)
+        vevent.add('description').value = self.submission.abstract or ""
+        vevent.add('url').value = self.submission.urls.public.full(scheme='https')
```

Comment:
Use the scheme out of `settings.SITE_URL` here. Everything should be https:// in 2017, but this is still a settings which should be honored here. Dev instances are http most of the time, for example.

Reply:
See other comment. Will happen in a site-wide class instead of in six (or so) separate places.